docs/modules/ROOT/pages/erc721.adoc

```diff
@@ -1,772 +1,361 @@
- ** <<ready_to_use_presets,Ready_to_Use Presets>>
-* <<extensibility,Extensibility>>
-* <<presets,Presets>>
- ** <<erc721mintableburnable,ERC721MintableBurnable>>
- ** <<erc721mintablepausable,ERC721MintablePausable>>
- ** <<erc721enumerablemintableburnable,ERC721EnumerableMintableBurnable>>
-  *** <<ierc721enumerable,IERC721Enumerable>>
- ** <<erc721metadata,ERC721Metadata>>
-  *** <<ierc721metadata,IERC721Metadata>>
-* <<utilities,Utilities>>
- ** <<erc721_holder,ERC721Holder>>
-* <<api_specification,API Specification>>
- ** <<ierc721_api,`IERC721`>>
-  *** <<balanceof,`balanceOf`>>
-  *** <<ownerof,`ownerOf`>>
-  *** <<safetransferfrom,`safeTransferFrom`>>
-  *** <<transferfrom,`transferFrom`>>
-  *** <<approve,`approve`>>
-  *** <<setapprovalforall,`setApprovalForAll`>>
-  *** <<getapproved,`getApproved`>>
-  *** <<isapprovedforall,`isApprovedForAll`>>
- ** <<events,Events>>
-  *** <<approval_event,`Approval (event)`>>
-  *** <<approvalforall_event,`ApprovalForAll (event)`>>
-  *** <<transfer_event,`Transfer (event)`>>
- ** <<ierc721metadata,`IERC721Metadata`>>
-  *** <<name,`name`>>
-  *** <<symbol,`symbol`>>
-  *** <<tokenuri,`tokenURI`>>
- ** <<ierc721enumerable,`IERC721Enumerable`>>
-  *** <<totalsupply,`totalSupply`>>
-  *** <<tokenbyindex,`tokenByIndex`>>
-  *** <<tokenofownerbyindex,`tokenOfOwnerByIndex`>>
- ** <<ierc721receiver_api,`IERC721Receiver`>>
-  *** <<onerc721received,`onERC721Received`>>
-
-== IERC721
-
-[,cairo]
+:token-types: https://docs.openzeppelin.com/contracts/4.x/tokens#different-kinds-of-tokens[non-fungible tokens]
+:eip721: https://eips.ethereum.org/EIPS/eip-721[EIP-721]
+:erc721-api: xref:/api/erc721.adoc[API Reference]
+:introspection: xref:/introspection.adoc[Introspection]
+
+The ERC721 token standard is a specification for {token-types}, or more colloquially: NFTs.
+The `token::erc721::ERC721` contract implements an approximation of {eip721} in Cairo for Starknet.
+
+TIP: For detailed information on the usage and implementation check the {erc721-api} section.
+
+== Interface
+
+:compatibility: xref:/erc721.adoc#erc721_compatibility[ERC721 Compatibility]
+:ierc721-interface: xref:/erc721.adoc#ierc721[IERC721]
+:ierc721metadata-interface: xref:/erc721.adoc#ierc721metadata[IERC721Metadata]
+:isrc5-interface: xref:/erc721.adoc#isrc5[ISRC5]
+
+ERC721 contracts must implement both the {ierc721-interface} and {isrc5-interface} interfaces in order to have an ERC721-compliant contract.
+Additionally, ERC721 contracts often include {ierc721metadata-interface} interface as well.
+The following preset contract includes all three interfaces and represents the full interface of the `ERC721ABI` module.
+
+WARNING: The {ierc721-interface} and {ierc721metadata-interface} interfaces are approximations of their Ethereum counterparts.
+These approximations include a few notable differences.
+See {compatibility} for more information.
+
+[,javascript]
+----
+trait IERC721ABI {
+    // IERC721
```

Comment:
now that we have components and presets, I wonder if we should begin with this chunk of interface. It's kindof big and with lots of information (non standard, camel support). maybe i'd focus in the component. what do you think?

Reply:
I agree that it is kind of big with a lot going on. I'm not sure what you're suggesting though. Do you mean removing the `Interface` section entirely? Or removing the individual interface subsections and just keep the `ERC721ABI`?

Reply:
i'm not sure i understood the second option, did you mean `IERC721`? i was suggesting to remove the interface section, although it may require a bit more of thought. My impression is that it's not that important now we have components because it's not a single module with all of it anymore, now you can select the impls.

If anything, it is closer to the preset, which is no yet documented and probably not the main focus of the section either.

Reply:
> i'm not sure i understood the second option, did you mean IERC721?

Yeah, that section, `IERC721Metadata`, and `ISRC5`. I removed those section already since I left that response

> i was suggesting to remove the interface section, although it may require a bit more of thought. My impression is that it's not that important now we have components because it's not a single module with all of it anymore, now you can select the impls.

> If anything, it is closer to the preset, which is no yet documented and probably not the main focus of the section either. The focus should be the component and design/implementation details.

Agreed^

Maybe it'd make sense to follow a structure like this in the doc:

`ERC721Component`
- `Implementations`
  - `ERC721Impl`
  - `ERC721MetadataImpl`
  ...
- `ERC721 compatibility`
- `Usage`
...

We can either head the `impl`s with the interface snippet or just provide a link to the interface in the API...or just remove the `Interface`/`Implementations` section entirely, idk

Reply:
Right now we have this inconsistent structures:

**Upgrades**
1. explainer
2. usage

**Account**
1. interface

**Access**
1. usage
2. interface

**ERC20**
1. interface
4. compat
5. usage

**ERC721**
2. interface
3. compat
4. usage

**Security**
1. usage

**Introspection**
1. interface
2. usage

Reply:
maybe we should have

1. explainer
2. usage
3. compatibility

and leave interface for API, of have it between explainer and usage